crates/circuit/src/operations.rs

```diff
@@ -286,6 +288,27 @@ impl Operation for OperationRef<'_> {
             Self::Unitary(unitary) => unitary.directive(),
         }
     }
+    #[inline]
+    fn get_mat_static_1q(&self, params: &[Param]) -> Option<[[Complex64; 2]; 2]> {
```

Comment:
Given that the current method to get the matrix is called `matrix`, how about calling these `matrix_as_static` and `matrix_as_nalgebra` for a consistent naming?

Reply:
This makes sense. The new names as `matrix_as_static_1q` and `matrix_as_nalgebra_1q` to emphasize that we only expect 1q matrices.